input/otlp/traces.go

```diff
@@ -837,10 +832,11 @@ func TranslateSpan(spanKind ptrace.SpanKind, attributes pcommon.Map, event *mode
 					event.DataStream = modelpb.DataStreamFromVTPool()
 				}
 				event.DataStream.Namespace = stringval
-
 			default:
-				modelpb.Labels(event.Labels).Set(k, stringval)
+				setLabel(k, event, v)
 			}
+		default:
```

Comment:
This is not equiv to ValueTypeSlice + pcommon.ValueTypeDouble. Are you relying on `setLabel` to filter out types e.g. Map? I'd prefer we keep the previous structure, as it is slightly easier to reason about the change, given how TranslateTransaction is structured a little differently to others e.g. TranslateSpan. The change now means both `switch` contains logic, while the previous code relies on the switch in TranslateTransaction, and the `switch` within setLabel is mostly a no-op. But I don't have a strong opinion about this.

Reply:
I just noticed I didn't mirror this change in `TranslateTransaction`. My mistake. My logic here is simply that, `setLabel` populates the `APMEvent` label map when no value derivations are needed. Meaning, if no special preprocessing needs to be done to "generate" an attribute value, then we can simple set the value in APMEvent directly (after OTLP -> Go primitive conversion, which is the purpose of `setLabel`). 

This follows from my comment in this PR description: _"To my limited knowledge, this is what I understand the purpose of setLabel to be; a producer of Go primitives when the exact translation is not made explicit."_

Said differently, with my proposed change, the `switch` statement would say the following:

> If we know the value to be set in APMEvent, default to setting it via `setLabel`. However, if we have to derive the value, do so in the appropriate switch-branch.

Reply:
In general, I see a `switch` branch without a `default` as an antipattern, since your "handling" is not exhaustive. This is something Rust has done very well with their `match` equivalent.